dynamic shapes: support randn.generator when generator is None

bdhirsh opened the pull request:
Fixes https://github.com/pytorch/pytorch/issues/107170

Today, we have a decomp for `randn.default` that handles dynamic shapes properly.

It looks like we don't have a decomp for the `randn.generator` overload of randn - but if you pass in a `None` generator, we can just call the default decomp. We should probably support the generator overload at some point (although I'm not sure what we do with generators in our IR today).

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116223

bdhirsh committed:
dynamic shapes: support randn.generator when generator is None

[ghstack-poisoned]

Skylion007 commented on torch/_refs/__init__.py:
```diff
@@ -5427,6 +5427,32 @@ def randn(
+) -> TensorLikeType:
+    # We should eventually support the generator overload.
+    # However, if someone passes in a None generator explicitly,
+    # we can jut fall back to randn.default
```
```suggestion
    # we can just fall back to randn.default
```

Skylion007 commented:
@pytorchbot rebase

bdhirsh committed:
Update on "dynamic shapes: support randn.generator when generator is None"

Fixes https://github.com/pytorch/pytorch/issues/107170

Today, we have a decomp for `randn.default` that handles dynamic shapes properly.

It looks like we don't have a decomp for the `randn.generator` overload of randn - but if you pass in a `None` generator, we can just call the default decomp. We should probably support the generator overload at some point (although I'm not sure what we do with generators in our IR today).




[ghstack-poisoned]

bdhirsh committed:
Update on "dynamic shapes: support randn.generator when generator is None"

Fixes https://github.com/pytorch/pytorch/issues/107170

Today, we have a decomp for `randn.default` that handles dynamic shapes properly.

It looks like we don't have a decomp for the `randn.generator` overload of randn - but if you pass in a `None` generator, we can just call the default decomp. We should probably support the generator overload at some point (although I'm not sure what we do with generators in our IR today).




[ghstack-poisoned]

bdhirsh commented:
@pytorchbot merge

bdhirsh commented:
Need to investigate CI (it roughly looks like dynamo isn't properly graph breaking on `randn.Generator` anymore when a generator is actually passed in)

aviator19941 commented:
@bdhirsh is there an update on merging this in? I've been keeping a manual aten.randn.generator decomposition in my codebase in the meanwhile, but was wondering if this was going to get merged in. Thank you very much!
